TauAnalysis/MCEmbeddingTools/plugins/MuMuForEmbeddingSelector.cc

```diff
@@ -32,21 +32,33 @@
 #include "DataFormats/Candidate/interface/CompositeCandidate.h"
 #include "DataFormats/PatCandidates/interface/Muon.h"
 
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+
+#include "DataFormats/METReco/interface/MET.h"
+#include "DataFormats/PatCandidates/interface/MET.h"
+
 //
 // class declaration
 //
 
 class MuMuForEmbeddingSelector : public edm::stream::EDProducer<> {
 public:
-  explicit MuMuForEmbeddingSelector(const edm::ParameterSet&);
+  explicit MuMuForEmbeddingSelector(const edm::ParameterSet &);
 
-  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
 
 private:
-  void produce(edm::Event&, const edm::EventSetup&) override;
+  void produce(edm::Event &, const edm::EventSetup &) override;
 
   // ----------member data ---------------------------
   edm::EDGetTokenT<edm::View<reco::CompositeCandidate>> ZmumuCandidates_;
+  edm::EDGetTokenT<reco::VertexCollection> theVertexLabel_;
+  edm::EDGetTokenT<reco::BeamSpot> theBeamSpotLabel_;
+  edm::EDGetTokenT<edm::View<pat::MET>> theMETLabel_;
+  edm::EDGetTokenT<edm::View<pat::MET>> thePuppiMETLabel_;
+  bool use_zmass = false;
   double ZMass = 91.0;
```

Comment:
That's a bit approximate, no?  Don't we want to use the exact value?

Reply:
Done in a8eae9f.